### PR TITLE
Setup: add requirement freeze for next major version

### PR DESCRIPTION
**Pull request by Borda**
## What does this PR do?

implement freezing version for next major requirement release

## Before submitting

- [x] Was this **discussed/approved** via a GitHub issue? (not for typos and docs)
- [x] Did you read the [contributor guideline](https://github.com/Lightning-AI/lightning/blob/master/.github/CONTRIBUTING.md), **Pull Request** section?
- [x] Did you make sure your **PR does only one thing**, instead of bundling different changes together?
- [ ] Did you make sure to **update the documentation** with your changes? (if necessary)
- [ ] Did you write any **new necessary tests**? (not for typos and docs)
- [ ] Did you verify new and **existing tests pass** locally with your changes?
- [ ] Did you list all the **breaking changes** introduced by this pull request?
- [ ] Did you **update the CHANGELOG**? (not for typos, docs, test updates, or minor internal changes/refactors)

## PR review

Anyone in the community is welcome to review the PR.
Before you start reviewing, make sure you have read the [review guidelines](https://github.com/Lightning-AI/lightning/wiki/Review-guidelines). In short, see the following bullet-list:

- [x] Is this pull request ready for review? (if not, please submit in draft mode)
- [x] Check that all items from **Before submitting** are resolved
- [x] Make sure the title is self-explanatory and the description concisely explains the PR
- [x] Add labels and milestones (and optionally projects) to the PR so it can be classified

## Did you have fun?

Make sure you had fun coding 🙃

cc @borda

### CODE REVIEW

**Review by manskx — APPROVED** [review]

File: .actions/setup_tools.py
Comment (hhsecond) [inline .actions/setup_tools.py:89]:
```suggestion
    # extract the major version from all listed versions
```

**Review by hhsecond — APPROVED** [review]

**Review by awaelchli — APPROVED** [review]

File: .actions/setup_tools.py
Comment (carmocca) [inline .actions/setup_tools.py:133]:
```suggestion
    '...PyTorch Lightning is just organized PyTorch...'
```

File: .actions/setup_tools.py
Comment (carmocca) [inline .actions/setup_tools.py:54]: I don't quite understand the difference between these. Can you explain it?

**Review by carmocca — APPROVED** [review]
This PR conflicts with https://github.com/Lightning-AI/lightning/pull/13912, which I feel would make this much easier if it was landed.

It would be nice if you could help me there with the CI failures that block it.

Additionally, can you update the PR title before merging?

Approving to unblock, but I would like to see #13912 be merged too

File: .actions/setup_tools.py
Comment (Borda) [reply .actions/setup_tools.py:133]: Well seems we do not have this tested for a while so let's leave it for another PR as we would also need to edit CI

Comment (carmocca) [reply .actions/setup_tools.py:133]: This suggestion is what I did in #13912. The doctest fails because william pushed a commit to master changing the beginning of the readme. My suggestion fixes the doctest

File: .actions/setup_tools.py
Comment (Borda) [reply .actions/setup_tools.py:54]: I think that the example help best to see the dirrence

**Comment by Borda** [comment]
> This PR conflicts with #13912, which I feel would make this much easier if it was landed.

This is needed for the 0.6 release on Monday

> It would be nice if you could help me there with the CI failures that block it.

yes, happy to help :)

File: .actions/setup_tools.py
Comment (Borda) [reply .actions/setup_tools.py:133]: addressing in #14498

**Pull request merged**
